test/cql-pytest/test_compactionstats.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
```

Comment:
I don't know why the test/topology guys started with the bad (in my opinion) tradition of having each tiny test function in a separate file, I think it's a mistake and encourages people not to think about writing related sets of tests and just thinking one test is enough, or not pausing to think how a specific test perhaps relates to other tests that already exist - or that will need to exist in the future.

Please try to think maybe what "area" this test might belong to and might need more tests. Will you maybe want more compaction tests and call it test_compaction? Maybe more "tasks" related tests?  Now that I think of it, maybe the existing test/rest_api/test_task_manager.py is the right place for this task? Can you please take a look?

Reply:
> I don't know why the test/topology guys started with the bad (in my opinion) tradition of having each tiny test function in a separate file, I think it's a mistake and encourages people not to think about writing related sets of tests and just thinking one test is enough, or not pausing to think how a specific test perhaps relates to other tests that already exist - or that will need to exist in the future.

I disagree with that sentiment, because it leads to what already happened in dtest and that is test files with thousands of lines, which are hard to navigate and comprehend. I would rather have a bunch of small files with individual tests or couple of tests, structured into modules. I think it is completely ok to have this test standalone, if we need additional compaction tests, we can create `cql-pytest/compaction` module and move this test there in the future.

Reply:
There are many bad things in dtest, the large test files was never one of them. As an example, dtest has a test file called materialized_views_test.py and it's obvious that you'll find all the the tests related to materialized views in there. Moreover, when you find a test inside that file, it's obvious it's about materialized views. Newcomers can also open that single file in an editor and browse through all the materialized views tests (each should have a comment explaining it, so this "browsing" usually means browsing through comments - not understanding each line of code).

Yes, people who programmed in Java many years (surprisingly, I'm one of them) don't like to have one large file and instead like a "module" (a subdirectory) with a lot of small files with a single function in it. In my opinion, doing this religiously (as the Java guys do) is a big mistake. There's a lot of benefits of storing multiple tests in a single file. The different tests of the same file can share the same fixtures (e.g., a common table only created once) or utility functions. They can share comments - and introduction to the whole set of tests, or an introduction to a few consecutive tests that need a common introduction.

As a random example, I invite you to take a look at test/cql-pytest/test_empty.py. This file has 20 small tests for one common theme - "empty" (zero-length) values. Most of the tests use a common fixture (making them faster), reading through this file, you can a sense of what is the unique problem of zero-length value, and perhaps even think what additional tests might be missing.

Another random example is test_json.py, with 48 (!) tests related to various aspects of JSON in Scylla - the fromJson() and toJson() functions, SELECT JSON, and INSERT JSON. Many of these tests share common utility functions and common test fixtures, and also related conceptually and have related comments (e.g., toJson() and SELECT JSON share the same bugs so often we have the same test in two variants).

In my opinion tiny test files with a single file - and moreover directories with just a few files - is wreaking havoc in the "topology" test framework. Among other things, I see it encourages developers to write incomplete tests (after all, if one test in a file is acceptible, why bother writing 20 tests about empty values? or 48 tests on JSON? nobody will complain if there are just one or two test functions). And it encourages developers not to share slow fixtures (like clusters) between different tests. 

Please note that the cql-pytest DOES allow subdirectories. We do have one prominent subdirectory (cassandra_tests, with translated Cassandra unit tests) and various subdirectories in it, and we can add more. But this PR did NOT propose adding a subdirectory with many small test files - it proposed adding just one single small test file.